src/dex/idle-dao/idle-dao.ts

```diff
@@ -0,0 +1,372 @@
+  // Rate changes almost every block.
+  // rpc-based pricing is only done for limited amount of tokens that are supported
+  async getTokenPrice(
+    idleToken: IdleToken,
+    blockNumber: number,
+  ): Promise<bigint | null> {
+    if (!idleToken.cdoContract) {
+      return null;
+    }
+
+    const tranchePrice = await idleToken.cdoContract.methods['virtualPrice'](
+      idleToken.idleAddress,
+    ).call({}, blockNumber);
+
+    return BigInt(tranchePrice);
+  }
+
+  // Returns estimated gas cost of calldata for this DEX in multiSwap
+  getCalldataGasCost(poolPrices: PoolPrices<IdleDaoData>): number | number[] {
+    return CALLDATA_GAS_COST.DEX_NO_PAYLOAD;
+  }
+
+  getAdapterParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: IdleDaoData,
+    side: SwapSide,
+  ): AdapterExchangeParam {
+    return {
+      // target exchange is not used by the contract
+      targetExchange: NULL_ADDRESS,
+      payload: '0x',
+      networkFee: '0',
+    };
+  }
+
+  getDexParam(
+    srcToken: Address,
+    destToken: Address,
+    srcAmount: NumberAsString,
+    destAmount: NumberAsString,
+    recipient: Address,
+    data: IdleDaoData,
+    side: SwapSide,
+    _: Context,
+    executorAddress: Address,
+  ): DexExchangeParam {
+    let returnAmountPos = undefined;
+
+    const [Interface, swapCallee, swapFunction, swapFunctionParams] = ((): [
+      Interface,
+      Address,
+      PoolFunctions,
+      Param,
+    ] => {
+      if (data.fromIdleToken) {
+        returnAmountPos = extractReturnAmountPosition(
+          this.cdo,
+          PoolFunctions[`withdraw${data.idleToken.tokenType}`],
+        );
+        return [
+          this.cdo,
+          data.idleToken.cdoAddress,
+          PoolFunctions[`withdraw${data.idleToken.tokenType}`],
+          [srcAmount],
+        ];
+      }
+
+      returnAmountPos = extractReturnAmountPosition(
+        this.cdo,
+        PoolFunctions[`deposit${data.idleToken.tokenType}`],
+      );
+      return [
+        this.cdo,
+        data.idleToken.cdoAddress,
+        PoolFunctions[`deposit${data.idleToken.tokenType}`],
+        [srcAmount, recipient],
+      ];
+    })();
+
+    const exchangeData = Interface.encodeFunctionData(
+      swapFunction,
+      swapFunctionParams,
+    );
+
+    return {
+      needWrapNative: this.needWrapNative,
+      dexFuncHasRecipient: false,
+      exchangeData,
+      targetExchange: swapCallee,
+      returnAmountPos: side === SwapSide.SELL ? returnAmountPos : undefined,
+    };
+  }
+
+  async getTopPoolsForToken(
+    tokenAddress: Address,
+    limit: number,
+  ): Promise<PoolLiquidity[]> {
+    if (!this.tokenList) {
+      const blockNumber =
+        await this.dexHelper.web3Provider.eth.getBlockNumber();
+      this.initializePricing(blockNumber);
+    }
+    const idleTokens: IdleToken[] = getPoolsByTokenAddress(tokenAddress);
+
+    return idleTokens
```

Comment:
I am not very aware of the current dex, but want to clarify if it makes sense to `idleAddress -> address` connectors?
Because as I can see now, it only returns new connectors for the usual tokens (`address` field in the list), but maybe it's worth to know that `address` is the connector for `idleAddress` as well? (for `idleAddress` return `address` connector)

Reply:
did additional logic to check for idle connectors